api/src/handlers/websocket/main.go

```diff
@@ -0,0 +1,79 @@
+package websocket
+
+import (
+	"github.com/gin-gonic/gin"
+	"github.com/gorilla/websocket"
+	"net/http"
+	"encoding/json"
+	"time"
+	"porter/api/src/lib/github"
+	"porter/api/src/handlers/websocket/message"
+	"fmt"
+	"porter/api/src/handlers/websocket/events"
+)
+
+var upgrader = websocket.Upgrader{
+	ReadBufferSize:  1024,
+	WriteBufferSize: 1024,
+	CheckOrigin: func(r *http.Request) bool {
+		return true
+	},
+}
+
+type ReceivedSocketEvent struct {
+	Type string `json:"type"`
+}
+
+func Handler(c *gin.Context) {
+	conn, err := upgrader.Upgrade(c.Writer, c.Request, nil)
+	if err != nil {
+		return
+	}
+
+	for {
+		_, msg, err := conn.ReadMessage()
+		if err != nil {
+			fmt.Println(err)
+			break
+		}
+§
+		var event ReceivedSocketEvent
+		err = json.Unmarshal(msg, &event)
+		if err != nil {
+			fmt.Println(err)
+			continue
+		}
+
+		go handlerSockerEvent(event, conn)
```

Comment:
opening a go routine cause I don't wanna stuck the connection

Reply:
Thats cool, but why are you then opening two more routines ?